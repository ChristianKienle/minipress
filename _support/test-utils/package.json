{
  "name": "@minipress/test-utils",
  "version": "0.7.0",
  "description": "",
  "files": [
    "src/**/*.js",
    "types/**/*.d.ts"
  ],
  "private": false,
  "publishConfig": {
    "access": "public"
  },
  "main": "src/index.js",
  "license": "MIT",
  "dependencies": {
    "fs-extra": "^8.1.0"
  },
  "typings": "types/index.d.ts"
}
