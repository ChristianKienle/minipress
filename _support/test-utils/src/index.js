// @ts-check
const TempDir = require('./temp-dir')

module.exports = {
  TempDir,
  createTempDir: () => new TempDir()
}
