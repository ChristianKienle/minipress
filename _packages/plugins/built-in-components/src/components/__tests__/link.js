import { shallowMount, createLocalVue } from '@vue/test-utils'
import Link from './../link.vue'
import VueRouter from 'vue-router'

const createLink = (options = {}) => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)
  const stubs = ['router-link', 'router-view']
  const mountOptions = {
    stubs,
    localVue,
    ...options
  }
  return shallowMount(Link, mountOptions)
}

describe('MiniLink', () => {
  it('renders internal links as router-link', () => {
    expect(createLink({ propsData: { to: '/' }})).toMatchSnapshot()
  })

  it('renders external links as anchor', () => {
    expect(createLink({ propsData: { to: 'https://example.org' }})).toMatchSnapshot()
  })
})
