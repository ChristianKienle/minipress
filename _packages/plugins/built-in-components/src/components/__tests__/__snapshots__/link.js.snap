// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniLink renders external links as anchor 1`] = `<a href="https://example.org" target="_blank" rel="noopener noreferrer"></a>`;

exports[`MiniLink renders internal links as router-link 1`] = `<router-link-stub to="/"></router-link-stub>`;
